vespajlib/src/test/java/com/yahoo/geo/OneDegreeParserTestCase.java

```diff
@@ -0,0 +1,266 @@
+        checkLat("S89\u00B059'59", -(89+59/60d+59/3600d));
+        checkLon("E179\u00B059'59", 179+59/60d+59/3600d);
+        checkLon("W179\u00B059'59", -(179+59/60d+59/3600d));
+    }
+
+    /**
+     * Tests inputs that are on latitude 90/-90 degrees and longitude 180/-180 degrees.
+     */
+    @Test
+    public void testOnBoundary() {
+        checkLat("N90", 90d);
+        checkLat("N90\u00B00'0", 90d);
+        checkLat("S90", -90d);
+        checkLat("S90\u00B00'0", -90d);
+
+        checkLon("E180", 180d);
+        checkLon("E180\u00B00'0", 180d);
+        checkLon("W180", -180d);
+        checkLon("W180\u00B00'0", -180d);
+    }
+
+    /**
+     * Tests inputs that are above latitude 90/-90 degrees and longitude 180/-180 degrees.
+     */
+    @Test
+    public void testAboveBoundary() {
+        String message = "";
+        try {
+            parser = new OneDegreeParser(false, "N90.0001");
+        } catch (IllegalArgumentException e) {
+            message = e.getMessage();
+        }
+        assertEquals("out of range [-90,+90]: 90.0001", message);
+        try {
+            parser = new OneDegreeParser(false, "S90.0001");
+        } catch (IllegalArgumentException e) {
+            message = e.getMessage();
+        }
+        assertEquals("out of range [-90,+90]: -90.0001", message);
+        try {
+            parser = new OneDegreeParser(true, "E180.0001");
+        } catch (IllegalArgumentException e) {
+            message = e.getMessage();
+        }
+        assertEquals("out of range [-180,+180]: 180.0001", message);
+        try {
+            parser = new OneDegreeParser(true, "W180.0001");
+        } catch (IllegalArgumentException e) {
+            message = e.getMessage();
+        }
+        assertEquals("out of range [-180,+180]: -180.0001", message);
+        try {
+            parser = new OneDegreeParser(false, "N90.000001");
+        } catch (IllegalArgumentException e) {
+            message = e.getMessage();
+        }
+        assertEquals("out of range [-90,+90]: 90.000001", message);
+        try {
+            parser = new OneDegreeParser(false, "S90.000001");
+        } catch (IllegalArgumentException e) {
+            message = e.getMessage();
+        }
+        assertEquals("out of range [-90,+90]: -90.000001", message);
+        try {
+            parser = new OneDegreeParser(true, "E180.000001");
+        } catch (IllegalArgumentException e) {
+            message = e.getMessage();
+        }
+        assertEquals("out of range [-180,+180]: 180.000001", message);
+        try {
+            parser = new OneDegreeParser(true, "W180.000001");
+        } catch (IllegalArgumentException e) {
+            message = e.getMessage();
+        }
+        assertEquals("out of range [-180,+180]: -180.000001", message);
+    }
+
+    /**
+     * Tests various inputs that contain syntax errors.
+     */
+    @Test
+    public void testInputErrors() {
+        String message = "";
+        try {
```

Comment:
Please add helper function to avoid code duplication. Similar as test above.

Reply:
fixed.